config/import-control.xml

```diff
@@ -57,29 +63,33 @@
 
   <subpackage name="api">
     <allow pkg="com.puppycrawl.tools.checkstyle.grammars"/>
-    <allow pkg="java.beans"/>
-    <allow pkg="java.text"/>
+    <allow pkg="java.lang.reflect" local-only="true"/>
+    <allow pkg="java.nio.charset" local-only="true"/>
+    <allow class="java.beans.PropertyDescriptor" local-only="true"/>
+    <allow class="java.text.MessageFormat" local-only="true"/>
     <allow class="com.puppycrawl.tools.checkstyle.grammars.CommentListener"
            local-only="true"/>
     <!-- AccessModifier is in util package (should be moved to api package) to disallow
     its usage by API clients till https://github.com/checkstyle/checkstyle/issues/3511-->
     <allow class="com.puppycrawl.tools.checkstyle.checks.naming.AccessModifier"
            local-only="true"/>
-    <allow class="com.puppycrawl.tools.checkstyle.grammars.GeneratedJavaTokenTypes"
-           local-only="true"/>
-    <allow class="com.puppycrawl.tools.checkstyle.Utils"
-           local-only="true"/>
+    <allow pkg="org.antlr.v4.runtime" local-only="true"/>
 
     <!-- allowed till https://github.com/checkstyle/checkstyle/issues/3455 -->
     <allow class="com.google.common.io.Closeables" local-only="true"/>
     <allow class="com.google.common.collect.ImmutableCollection" local-only="true"/>
     <allow class="com.google.common.collect.ImmutableMap" local-only="true"/>
+
+    <allow class="com.puppycrawl.tools.checkstyle.Checker"/>
+    <!-- allowed till https://github.com/checkstyle/checkstyle/issues/3817 -->
+    <allow pkg="com.puppycrawl.tools.checkstyle.utils"/>
   </subpackage>
 
   <subpackage name="checks">
     <allow pkg="java.math"/>
-    <allow pkg="com.puppycrawl.tools.checkstyle.checks"/>
-    <allow class="com.puppycrawl.tools.checkstyle.Definitions"/>
+    <allow pkg="com.puppycrawl.tools.checkstyle"/>
```

Comment:
it sounds like alot ... can we restrict usage of other packages as much as possible ?